test/nidmExampleDataTest.m

```diff
@@ -6,44 +6,49 @@
 classdef nidmExampleDataTest < matlab.unittest.TestCase
     
     methods(TestMethodSetup)
+        
         %Rename the users HTML folder to prevent the tests damaging it.
         function storeUsersHTML(testCase)
-            if exist(fullfile(fileparts(mfilename('fullpath')), '..', 'Data', 'index.html'), 'file') == 2
-                movefile(fullfile(fileparts(mfilename('fullpath')), '..', 'Data', 'index.html'), fullfile(fileparts(mfilename('fullpath')), '..', 'Data', 'indexTemp.html'))
-            end
+            storeFile(fullfile(fileparts(mfilename('fullpath')), '..', 'Data', 'fsl_default'));
+            storeFile(fullfile(fileparts(mfilename('fullpath')), '..', 'Data', 'ex_spm_default'));
+            storeFile(pwd); 
         end
+        
     end
-    
-    methods(TestMethodTeardown)
-        %Remove the HTML folder created by test and move the users data back to the HTML folder.
-        function removeTestHTML(testCase)
-            if exist(fullfile(fileparts(mfilename('fullpath')), '..', 'Data', 'index.html'), 'file') == 2
-                delete(fullfile(fileparts(mfilename('fullpath')), '..', 'Data', 'index.html'))
-            end
-            if exist(fullfile(fileparts(mfilename('fullpath')), '..' , 'Data', 'indexTemp.html'), 'file') == 2
-                movefile(fullfile(fileparts(mfilename('fullpath')), '..', 'Data', 'indexTemp.html'), fullfile(fileparts(mfilename('fullpath')), '..', 'Data', 'index.html'), 'f')
-            end
-        end
-    end 
         
     methods(Test)
-        %Simply checking the viewer doesn't crash.
-        function checkViewerRuns(testCase)
-            nidm_results_display(fullfile(fileparts(mfilename('fullpath')), '..', 'Data', 'nidm.json'), true);
+        
+        %Checking the viewer runs on SPM-nidm input.
+        function checkViewerRunsSPM(testCase)
+            nidm_results_display(fullfile(fileparts(mfilename('fullpath')), '..', 'Data', 'ex_spm_default', 'nidm.json'));
         end
+        
         %Checking the experiment title is somewhere in the output HTML
         %file.
         function checkForTitle(testCase)
-            nidm_results_display(fullfile(fileparts(mfilename('fullpath')), '..', 'Data', 'nidm.json'), true);
-            text = fileread(fullfile(fileparts(mfilename('fullpath')), '..', 'Data', 'index.html'));
+            nidm_results_display(fullfile(fileparts(mfilename('fullpath')), '..', 'Data', 'ex_spm_default', 'nidm.json'));
```

Comment:
What we could do to avoid re-running the viewer, for each test. Is to have one test class per test dataset and then the call to `nidm_results_display` would be part of the setup and then each test is a new method.